src/lib/GenericPredictor.cc

```diff
@@ -4,17 +4,25 @@
 
 namespace simeng {
 
-GenericPredictor::GenericPredictor(YAML::Node config)
-    : btbBits_(config["Branch-Predictor"]["BTB-Tag-Bits"].as<uint64_t>()),
-      btb_(1 << btbBits_,
-           {config["Branch-Predictor"]["Fallback-Static-Predictor"]
-                .as<uint16_t>(),
-            0}),
-      satCntBits_(
-          config["Branch-Predictor"]["Saturating-Count-Bits"].as<uint64_t>()),
-      globalHistoryLength_(
-          config["Branch-Predictor"]["Global-History-Length"].as<uint64_t>()),
-      rasSize_(config["Branch-Predictor"]["RAS-entries"].as<uint64_t>()) {
+GenericPredictor::GenericPredictor() {
+  ryml::ConstNodeRef config = config::SimInfo::getConfig();
+  btbBits_ = config["Branch-Predictor"]["BTB-Tag-Bits"].as<uint8_t>();
```

Comment:
There seems to be some mixing and matching regarding if `config` is passed as a constructor / function argument (with default value set to `config::SimInfo::getConfig()`) or if it is fetched at the start of the constructor / function.

For example, Linux process passes config as an argument with a default value, but generic predictor fetches config at the start of the function.

Passing config as an argument with a defualt value does allow for the initialiser list to be utilised more, so I would probably have this method as our goto